sunxspex/xspec_models.py

```diff
@@ -0,0 +1,367 @@
+        internal_flux=np.zeros(photon_energies.shape)
+
+        # Constants
+        #const=constants.Constants() #assume this has already happened
+        mc2 = const.get_constant('mc2')
+        clight = const.get_constant('clight')
+        au = const.get_constant('au')
+        r0 = const.get_constant('r0')
+
+        # Max number of points
+        maxfcn = 2048
+
+        # Average atomic number
+        z = 1.2
+
+        # Relative error
+        rerr = 1e-4
+
+        # Numerical coefficient for photo flux
+        fcoeff = ((clight ** 2 / mc2 ** 4) / (4 * np.pi * au ** 2))
+        decoeff = 4.0 * np.pi * (r0 ** 2) * clight
+
+        # Create arrays for the photon flux and error flags.
+        iergq = np.zeros_like(photon_energies, dtype=np.float64)
+        try:
+            a0,p, eebrk, q, eelow, eehigh,_=params #why is this not consistent within xspec?
+        except ValueError:
+            a0,p, eebrk, q, eelow, eehigh=params
+
+        #if eelow >= eehigh:
+        #    return list(photon_energies)[1:]
+
+        i, = np.where(np.logical_and(photon_energies < eehigh, photon_energies > 0))
+
+        if i.size > 0:
+            try:
+                internal_flux[i], iergq[i] = split_and_integrate(model='thick-target', photon_energies=photon_energies[i],maxfcn=maxfcn, rerr=rerr, eelow=eelow,eebrk=eebrk, eehigh=eehigh, p=p, q=q, z=z,efd=False)
+            except ValueError:
+                return flux #same as previous iteration result... might not always want this however
+
+            internal_flux = (fcoeff / decoeff) * internal_flux
+
+            #logging.info(f"{dt.now()} PARAMS {params}")
+
+            #interpolate to bin centers as proxy for integral...for now
+            f=interp1d(photon_energies,internal_flux)
+            internal_flux=f(photon_energy_bin_centers)*a0*1e35
+
+            #have to modify inplace, not return another pointer
+            flux[:]=[internal_flux[j] if j in i else prev for j,prev in enumerate(flux)]*photon_energy_bins
+
+class ThickTargetModel0(XspecModel):
+    '''Thick-target bremsstrahlung model for use in Xspec. Uses original IDL-based integration. Default parameters are taken from OSPEX [link]()'''
+    def __init__(self):
+        self.ParInfo=(
+        "a0  1e-35  100.0  1.0  10.0  1e6  1e7  1.0",
+        "p  \"\"  4.0  1.1  1.5  15.0  20.0  0.1",
+           "eebrk  keV  150.0  1.0  5.0  100.  1e5  0.5" ,
+         "q  \"\"  6.0  0.0  1.5  15.0  20.0  0.1",
+         "eelow  keV  20.0  0.0  1.0  100.  1e3  1.0" ,
+         "eehigh  keV  3200.0  1.0  10.0  1e6  1e7  1.0"
+        ) #default parameters from OSPEX
+        self.model=self.thick2_original
+        self.description=f"Thick-target bremsstrahlung '{self.model.__name__}'"
+
+    @staticmethod #XSPEC won't know what to do with self as first arg
+    def thick2_original(photon_energies, params, flux):
```

Comment:
If this doesn't work, should it just be deleted?

Reply:
can definitely get rid of this, it was there mainly for testing purposes